verification/VerifyImputation.wdl

```diff
@@ -0,0 +1,162 @@
+        input:
+          file1 = select_first([single_sample_test_vcf])[idx],
+          file2 = select_first([single_sample_truth_vcf])[idx]
+      }
+    }
+  }
+
+  call CrosscheckFingerprints {
+    input:
+      firstInputs = if (defined(input_multi_sample_vcf)) then select_all([input_multi_sample_vcf]) else select_first([input_single_sample_vcfs]),
+      firstInputIndices = if (defined(input_multi_sample_vcf_index)) then select_all([input_multi_sample_vcf_index]) else select_first([input_single_sample_vcfs_indices]),
+      secondInputs = [test_vcf],
+      secondInputIndices = [test_vcf_index],
+      haplotypeDatabase = haplotype_database
+  }
+
+  if (split_output_to_single_sample) {
+    call ImputationTasks.SplitMultiSampleVcf {
+      input:
+        multiSampleVcf = test_vcf
+    }
+
+    call CrosscheckFingerprints as CrosscheckFingerprintsSplit {
+      input:
+        firstInputs = if (defined(input_multi_sample_vcf)) then select_all([input_multi_sample_vcf]) else select_first([input_single_sample_vcfs]),
+        firstInputIndices = if (defined(input_multi_sample_vcf_index)) then select_all([input_multi_sample_vcf_index]) else select_first([input_single_sample_vcfs_indices]),
+        secondInputs = SplitMultiSampleVcf.single_sample_vcfs,
+        secondInputIndices = SplitMultiSampleVcf.single_sample_vcf_indices,
+        haplotypeDatabase = haplotype_database
+    }
+  }
+
+  output {
+  }
+  meta {
+    allowNestedInputs: true
+  }
+}
+
+task CompareImputationMetrics {
+  input {
+    File test_metrics
+    File truth_metrics
+  }
+  command <<<
+  set -eo pipefail
+  diff "~{test_metrics}" "~{truth_metrics}"
+
+  if [ $? -ne 0 ];
+  then
+    echo "Error: ${test_metrics} and ${truth_metrics}  differ"
+  fi
+  >>>
+
+  runtime {
+    docker: "ubuntu:20.04"
+    cpu: 1
+    memory: "3.75 GiB"
+    disks: "local-disk 10 HDD"
+  }
+}
+
+task CrosscheckFingerprints {
+  input {
+    Array[File] firstInputs
+    Array[File] secondInputs
+    Array[File] firstInputIndices
+    Array[File] secondInputIndices
+    File haplotypeDatabase
+    String basename = "plumbing_test"
+    String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.1.9.0"
+  }
+
+  Int disk_size = ceil(1.2*(size(firstInputs, "GiB") + size(secondInputs, "GiB") + size(haplotypeDatabase, "GiB"))) + 100
+
+  command <<<
+    # add links to ensure correctly located indices
+    array_vcfs=( ~{sep=" " firstInputs} )
+    array_indices=( ~{sep=" " firstInputIndices} )
+    for i in ${!array_vcfs[@]}; do
+      ln -s ${array_indices[i]} $(dirname ${array_vcfs[i]})
+    done
+
+    array_vcfs2=( ~{sep=" " secondInputs} )
+    array_indices2=( ~{sep=" " secondInputIndices} )
+    for i in ${!array_vcfs2[@]}; do
+      ln -s ${array_indices2[i]} $(dirname ${array_vcfs2[i]})
+    done
+
+    gatk CrosscheckFingerprints -I ~{sep=" -I " firstInputs} -SI ~{sep=" -SI " secondInputs} -H ~{haplotypeDatabase} -O ~{basename}.crosscheck
```

Comment:
CrossCheckFingerprints is really a picard tool.  Just a nit, but might want to use picard rather than gatk here.

Reply:
This was code that was ripped out of the Imputation.wdl/ImputationTasks.wdl and just moved over here. I am noticing the Imputation pipeline really doesn't use anything from picard, not sure if it was intentional or not. I can update it here to use picard!